core/src/main/java/bio/terra/pearl/core/service/site/LocalizedSiteContentService.java

```diff
@@ -60,6 +64,16 @@ public LocalizedSiteContent create(LocalizedSiteContent localSite) {
             NavbarItem savedItem = navbarItemService.create(navItem);
             savedSite.getNavbarItems().add(savedItem);
         }
+
+        for (int i = 0; i < localSite.getLanguageTextOverrides().size(); i++) {
```

Comment:
I don't think you need the index, so you can just use the regualr iteration
```suggestion
        for (LanguageText langText : localSite.getLanguageTextOverrides()) {
```

Reply:
It would be nice if these LanguageTexts also had the portalId set on them.  Not sure what's the best way to do that given that the LocalizedSiteContent doesn't have a portal Id to reference.

Reply:
I'm not sure the cleanest way to do this and I also worry it muddies up the model because right now, if portal is specified, then that language text is always global to the portal always, whereas if we specified both then the SQL would have to do a bunch of null/not null checks. Maybe that's the correct thing to do, but i do like the simplicity of them being "attached" to the portal in two different ways - globally or locally.